raft.go

```diff
@@ -652,6 +702,75 @@ func (r *Raft) verifyLeader(v *verifyFuture) {
 	}
 }
 
+// leadershipTransfer is doing the heavy lifting for the leadership transfer.
+func (r *Raft) leadershipTransfer(future *leadershipTransferFuture) {
+
+	// Step 1: set this field which stops this leader from responding to any client requests.
+	r.leaderState.leadershipTransferInProgress = true
+
+	// Step 2: make sure the target server is up to date
+	s, ok := r.leaderState.replState[future.ID]
+	if !ok {
+		err := fmt.Errorf("cannot find replication state for %v, aborting leadership transfer", future)
+		r.logger.Printf("[DEBUG] raft: %v", err)
+		future.respond(err)
+		return
```

Comment:
You are right we need to add `r.leaderState.leadershipTransferInProgress = false` here. I don't think we can defer that since the happy path needs it to be enabled. Because the target becomes leader after we returned from this function and we cannot accept client requests in the meanwhile.